src/execution/execute.js

```diff
@@ -108,6 +107,23 @@ export type ExecutionResult = {
   data?: ObjMap<mixed>,
 };
 
+/**
+ * A partial result of GraphQL execution. `data` and `errors` as above, but
+ * tuple to avoid allocating/destroying objects. "NP" = "Not Promise".
```

Comment:
Not sure if it matters (requires profiling) but a tuple is just an array, so will also incur allocation & GC.

Reply:
True. This was an experiment to see whether *objects* were more expensive than tuples, and the difference (according to my benchmark) is negligible!